src/edu/csus/ecs/pc2/core/Utilities.java

```diff
@@ -616,8 +618,29 @@ public static String getReportFilename(IReport selectedReport) {
      * @throws FileNotFoundException
      */
     public static String createReport(IReport report, IInternalContest contest, IInternalController controller, boolean printHeaderAndFooter) throws FileNotFoundException {
+        return createReport(Constants.REPORT_DIRECTORY_NAME, report, contest, controller, printHeaderAndFooter);
+    }
+    
+    /**
+     * Create/Write a report to file in outputDirectoryName, creates outputDirectoryName if dir doesnot exist. 
+     * 
+     * 
+     * @param outputDirectoryName output directory name
+     * @param report
+     *            IReport to write
+     * @param contest
+     * @param controller
+     * @return name of created report file.
+     * @throws FileNotFoundException
+     */
+    public static String createReport(String outputDirectoryName, IReport report, IInternalContest contest, IInternalController controller, boolean printHeaderAndFooter) throws FileNotFoundException {
 
         String filename = getReportFilename(report);
+        
+        if (outputDirectoryName != null && outputDirectoryName.trim().length() > 0) {
+            ExecuteUtilities.ensureDirectory(outputDirectoryName);
+            filename = outputDirectoryName + File.separator + filename;
+        }
```

Comment:
Since the above `if` block takes care to test that `outputDirectoryName` is not null and has characters in it,  it seems like there should also be something to handle the possibility that these conditions are NOT met (in other words, there should be an `else` clause).  Otherwise, what happens if `outputDirectoryName` is null?   (Right now I believe that it will end up with a value in `filename` that is like "`null/<filename>`" because the String concatenation operators in the line
```
 filename = outputDirectoryName + File.separator + filename;
```
will convert the `null` outputDirectoryName object to the string "null".  (I could be wrong about that, but in any case what _should_ the code do in the case of a null `outputDirectoryName`?)

Reply:
I think I misread the code when I made this comment.  Disregard.